httpcore5/src/main/java/org/apache/hc/core5/reactor/ssl/SSLIOSession.java

```diff
@@ -345,6 +347,13 @@ private void updateEventMask() {
         }
         if (this.status == CLOSING && !this.outEncrypted.hasData()) {
             this.sslEngine.closeOutbound();
+            final int n = this.outboundClosedCount.incrementAndGet();
+            // This is to avoid an infinite session spin in case
+            // SSLEngine#closeOutbound() fails to trigger a graceful
+            // session shutdown
+            if (n > 5) {
```

Comment:
Stupid question, where does the `5` come from? Why isn't this a static final int?

Reply:
@michael-o Out of my nose, I must admit. I have no idea how many repetitions may imply a session spin and a problem with the close notify handshake. Just two looked a bit too risky, 10 looked too much, so I ended up choosing 5.

Reply:
OK, maybe this should be really a static variable with a simple comment?!

Reply:
That makes sense, and if the value is arbitrary maybe it should be defined through `java.lang.Integer.getInteger(String, int)`...

Reply:
@michael-o What good would that make? This value is not supposed to be reused anywhere, so I see little value in externalizing it. What I can do is mention in the comment that 5 is completely arbitrary.

Reply:
I don't strive for reusabliity, but simply for documentation. One might consider the spot as code smell. A comment above would be sufficient. If necessary, it can be refactored afterwards.

Reply:
@michael-o I will see if I can come up with a better fix.